Feature/mark completely indexed

johannesdewit opened the pull request:
## Veranderingen

- Voegt `index-status = complete` toe aan compleet geïndexeerde werken in `works.yml`.
- Veranderd de `work-detail.html` pagina en laat een icoontje zien die de `index-status` aangeeft en toelicht.

https://user-images.githubusercontent.com/77152231/210751917-0be034c7-d39b-4286-986d-a3e102eefcd9.mov

## Waarom?

Het is zo duidelijk voor de gebruiker of de index die hij bekijkt volledig is of niet. Op basis daarvan kan de informatie op waarde worden geschat.

## Verdere ideeën

- Eventueel kunnen later nog auteurs van indices van werken worden weergegeven.
- Ook zouden we gedeeltes van werken eventueel kunnen aanmerken als ‘volledig geïndexeerd’.

johannesdewit committed:
Set up a file to track completely indexed works

johannesdewit committed:
Add G to completely-indexed

johannesdewit committed:
Add complete field and populate it from `completely-indexed.yml`

johannesdewit committed:
Add check icon to pages of completely indexed works

johannesdewit committed:
Restore order in the realms of migrations and tailwind

Remove extra line from faulty merge

andredelft commented on heidegger_index/templates/_complete_icon.html:
```diff
@@ -0,0 +1,7 @@
+<div title="This work has been completely indexed and reviewed." class="relative z-1 inline-flex align-top w-6 h-6 ml-0 duration-300 rounded-full text-slate-700 dark:text-slate-100 drop-shadow dark:drop-shadow-md transition-color hover:bg-slate-200 dark:hover:bg-slate-600">
```
Tip: `text-green-600` ipv `text-slate-700 dark:text-slate-100`.

andredelft commented on heidegger_index/templates/work_detail.html:
```diff
@@ -6,7 +6,7 @@
 {% include '_home_button.html' %}
 
-{% include 'components/h1.html' with text=work.csl_json.title class="clear-right italic" only %}
+{% include 'components/h1.html' with text=work.csl_json.title class="clear-right italic" complete=work.complete only %}
```
Liever niet aan `h1.html` doorgeven, die was generiek bedoeld. Strikt gezien is een `<h1>` ook alleen bedoeld voor tekst. Je kunt dit doen:

```html
<div class="flex">
  {% include 'components/h1.html' with text=work.csl_json.title class="clear-right italic" only %}
  {% if complete %}{% include '../_complete_icon.html' %}{% endif %}
</div>
```
  
```

andredelft commented on index/completely-indexed.yml:
```diff
@@ -0,0 +1,21 @@
+# WRK:
+#   complete: False
+#   index_author:
+#   - family: Doe
+#     given: John
+#   - family: Doe
+#     given: Jane
+#   sections_complete:
+#   - end: 32
+#     start: 25
+#   - start: 20
+
+G:
+  complete: True
+  index_author:
+  - family: Delft
+    given : André
+    dropping-particle: van
+  - family: Wit
+    given: Johannes
+    dropping-particle: de
```
Het voelt niet helemaal lekker om voor zo'n 'kleine' feature een aparte file te gebruiken. Volgens mij kun je deze velden best toevoegen aan objecten van `works.yaml`, de citeproc server negeert volgens mij gewoon de velden die niet bij de CSL spec horen.

Dan zou overigens `complete` misschien wel een wat onduidelijke naam zijn. `completely_indexed` wellicht dan gewoon. (Of `index_status`, voor als we in het vervolg nog andere statussen willen toevoegen..)

johannesdewit replied:
Ik zal ze gewoon in works.yml zetten. Ik had ze in een aparte file gezet omdat het niet CSL was, maar als de server ze gewoon negeerd kan dat inderdaad prima!

andredelft commented on heidegger_index/migrations/0010_work_complete.py:
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 4.1 on 2023-01-05 09:46
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('heidegger_index', '0009_lemma_gnd_id'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='work',
+            name='complete',
+            field=models.BooleanField(default=False),
+        ),
+    ]
```
Let even op dat er nu twee migration files van 0009 afhankelijk zijn. Ik zou als ik jou was in #25 even de migration files combineren (0009 en 0010 verwijderen en opnieuw een `makemigrations` doen).

andredelft reviewed (changes requested):
Doe nog even een review request als je #25 gemerged hebt, dan kan ik wat beter zien wat de wijzigingen van deze PR nou zijn. En in het vervolg kun je denk ik best de PR's onafhankelijk laten zijn, als je, wanneer je op het punt staat te mergen, de migration files even opnieuw bouwt. (Je ziet ook dat ze nu alsnog door elkaar zijn gelopen.)

johannesdewit committed:
Rewrite `work_detail.html` h1 with complete icon

Change completely index icon color

Display information icon when work is marked as incomplete.

johannesdewit committed:
Change model name + migration cleanup

Delete seperate model field for `index-status`

`index-status` is now in the `csl-json`.

johannesdewit committed:
Small cleanup

johannesdewit committed:
Delete `completely-indexed.yml`
